### PR TITLE
mempool p2p configuration and some cleanup

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: src/Chainweb/Mempool/Mempool.hs [hunk 1]
```diff
@@ -241,44 +242,69 @@ data SyncState = SyncState {
...
+            res <- (`V.zip` hashes) <$> mempoolMember localMempool hashes
+            let !newMissing = V.map snd $ V.filter (not . fst) res
+            let !newMissingCnt = V.length newMissing
+
+            -- Collect set of all remote hashes
+            let !remoteHashes' = V.foldl' (flip HashSet.insert) remoteHashes hashes
+
+            -- Count number of missing hashes and decide if there @tooMany@
             let !newCnt = cnt + fromIntegral newMissingCnt
             let !tooMany' = (newCnt >= maxCnt)
```
Comment (marklnichols) [inline src/Chainweb/Mempool/Mempool.hs]: You can drop the paren here

File: src/Chainweb/Mempool/Mempool.hs [hunk 2]
```diff
@@ -296,47 +322,44 @@ syncMempools' log0 us localMempool remoteMempool onInitialSyncComplete =
...
+        (SyncState numMissingFromLocal missingChunks remoteHashes _) <- readIORef syncState
+
+        deb $ T.concat
+            [ sshow (HashSet.size remoteHashes)
+            , " hashes at remote ("
+            , sshow numMissingFromLocal
+            , " need to be fetched)"
+            ]
+
+        numPushed <- fmap snd $ concurrently
```
Comment (marklnichols) [inline src/Chainweb/Mempool/Mempool.hs]: I'd probably use: snd <!> concurrently

File: src/Chainweb/Mempool/Consensus.hs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE BangPatterns #-}
```
[the remark this answers is not in the document]
Comment (larskuhtz) [reply src/Chainweb/Mempool/Consensus.hs]: My compiler was complaining that it was missing.

File: src/Chainweb/Mempool/Mempool.hs (hunk 1 above)
Comment (larskuhtz) [inline src/Chainweb/Mempool/Mempool.hs]:
```suggestion
            let !tooMany' = newCnt >= maxCnt
```

**Pull request merged**